types/underscore/underscore-tests.ts

```diff
@@ -2259,6 +2265,11 @@ undefinedIdentityIterateeResult; // $ExpectType StringRecord
     _.object(level2UnionList); // $ExpectType Dictionary<any>
     _(level2UnionList).object(); // $ExpectType Dictionary<any>
     extractChainTypes(_.chain(level2UnionList).object()); // $ExpectType ChainType<Dictionary<any>, any>
+
+    // non-nested lists
+    _.object(stringRecordList); // $ExpectType Dictionary<never>
```

Comment:
I think I might prefer `$ExpectError`. Is that feasible?

Reply:
It is feasible for `UnderscoreStatic` specifically if I switch to using multiple overloads, but I don't think it's feasible for `Underscore` or `_Chain` unfortunately since the list is not a parameter in those interfaces.

I went with this because it seemed like it would be nice to keep the behavior of all three of these the same and it seemed like it might best follow the [Do's and Don'ts guidance on optional parameters](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-optional-parameters). However, there are two things that are now making me think that separate overloads would in fact be better. First, it would admittedly likely be better to use the separate overload approach for the long-term vision of making `Underscore` and `_Chain` from `UnderscoreStatic` since if [this approach](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45201#issuecomment-637914857) were employed and worked like I'd hope then the tighter constraints on each of these (`List<EnumerableKey>` and `List<List<any>>`) would ideally cause these functions to be `never`ed out of existence for wrapped types that don't meet those constraints, which would make all of these error out in the future. Second, the Do's and Don'ts page does say that its guidance doesn't apply when return types for overloads differ. With return type capabilities like `_ObjectValue` and `IterateeResult` available in TS2.8+ that advice seems a little more murky to me, but I think the ideal difference in constraints here and the iffy `TValueList = undefined` I'm needing to do in type arguments to make this work crosses a line here that makes separate overloads better.

Reply:
More generally, the Do's and Don'ts page states that you shouldn't create multiple overloads (but use optional parameters instead) if *the only difference is in the trailing parameters*. But even if the return type is the same, which arguably, it can be, this is not the case here: to the contrary, the types of the first parameter in the two signatures are completely disjoint.

(Just making this point for additional justification of the move you already made.)